### ISSUE TITLE
MSRV

### ISSUE THREAD

**Issue by waywardmonkeys** [googlefonts/fontations#1019]
What is the desired MSRV for this family of crates?

Right now, it is 1.76 to get a build. This is because tests within `int-set` import `std::hash::DefaultHasher` which is a new location for it. Prior to 1.76, it was available  as `std::collections::hash_map::DefaultHasher`.

If that's addressed, then the MSRV is 1.75 because code within `int-set` uses the features described here: https://github.com/rust-lang/rust/issues/91611

**Comment by cmyr** [comment]
I think it's fine to say 1.76 for now, and I would not place much value on trying to maintain a low MSRV for the foreseeable future; that is, if something in a new rust release is useful I'm happy for us to adopt it.

**Comment by rsheeter** [comment]
Pragmatically it's the lowest of google3 and Chrome. I'm not sure that is
published in a form we could readily republish or consume in CI (which
would be nice to do at some point).

**Comment by waywardmonkeys** [comment]
Do you know offhand what that value is?

**Comment by nyurik** [comment]
The currently used 1.75 is a bit dated (December 2023). There has been quiet a few improvements since then - most are fairly small but together adding up. For example, c-literals `c"foo"` in v1.77 - not worth by itself, but cleaner for any FFI. If not a hard requirement, would it perhaps make sense to keep things "about a year old", i.e. 1.81 or so?   Or better yet, just switch to the 2024 edition (1.85)

**Comment by rsheeter** [comment]
As we now set a rust-version I believe this can be closed? - please reopen if I am mistaken.

**Comment by rsheeter** [comment]
Codegen is not something a user would depend on. It's fair to say that makes workspace level version misleading.

I agree the c-literal is much nicer but ... worth increasing MSRV? That i'm less sure of, as you say it seems like a nice to have.

**Comment by rsheeter** [comment]
Good catch re klippa, filed https://github.com/googlefonts/fontations/issues/1594.

I believe we did do work to check msrv in CI (afk so not checking rn), but likely only for the crates that are actively used by major consumers (font-types, read-fonts, skrifa). klippa is on it's way but not quite there yet. If that's a figment of my imagination then we'll add it :)

Ty for the reference re MSRV validation.
